_test/kind/setup.sh

```diff
@@ -0,0 +1,108 @@
+  echo "### Wait for Ingress controller to install ###"
+  kubectl wait --namespace ingress-nginx \
+    --for=condition=ready pod \
+    --selector=app.kubernetes.io/component=controller \
+    --timeout=90s
+  # Would like to find a cleaner approach to configure the podTemplate and Jenkins job below
+  TPL_TEMP=$(mktemp -d)
+  JENKINS_AGENT="${AGENT}" envsubst < ${SCRIPT_DIR}/jenkins-podtemplate.yaml > ${TPL_TEMP}/podtemplate.yaml
+  JENKINS_AGENT="${AGENT}" JENKINSFILE=$(sed '2,$s/^/                      /' ${AGENT_PATH}/Jenkinsfile.test) envsubst < ${SCRIPT_DIR}/jenkins-casc-config-scripts-template.yaml > ${TPL_TEMP}/jenkins-casc-config-scripts.yaml
+  # Use Helm to deploy and configure Jenkins
+  helm repo add jenkinsci https://charts.jenkins.io --force-update
+  helm repo update
+  helm install jenkins \
+    --version ${JENKINS_CHART_VERSION} \
+    -n jenkins --create-namespace \
+    -f ${SCRIPT_DIR}/jenkins-values.yaml \
+    -f ${TPL_TEMP}/podtemplate.yaml \
+    -f ${TPL_TEMP}/jenkins-casc-config-scripts.yaml \
+    jenkinsci/jenkins
+  # Make sure Jenkins is available 
+  echo "### Wait for Jenkins instance to become ready ###"
+  until [[ $(curl -s -w %{http_code} http://localhost/login -o /dev/null) == "200" ]]
+  do
+    sleep 5
+  done
+
+  # Create a Jenkins api token 
+  secret=$(kubectl get secret -n jenkins jenkins -o jsonpath='{.data.jenkins-admin-password}' | base64 -d)
+  crumb=$(curl -sk https://localhost/crumbIssuer/api/json --user admin:${secret} --cookie-jar /tmp/cookies | jq -r '.crumb')
+  token=$(curl -sk https://localhost/me/descriptorByName/jenkins.security.ApiTokenProperty/generateNewToken --data 'newTokenName=foo' --user admin:${secret} -H "Jenkins-Crumb: ${crumb}" --cookie /tmp/cookies | jq -r '.data.tokenValue')
+
+  # Start and monitor build
+  echo "Starting build for ${AGENT}..."
+  curl -s -XPOST http://localhost/job/containers-quickstarts/job/${AGENT}/build --user admin:${token}
+  timeout=0
+  until [[ $(curl -s http://localhost/job/containers-quickstarts/job/${AGENT}/lastBuild/api/json --user admin:${token} -w %{http_code} -o /dev/null) == "200" ]]
+  do
+    if [[ ${timeout} -gt 60 ]]
+    then
+      echo "Timed out waiting for build to be created..."
+      exit 1
+    fi
+    sleep 2
+    let "timeout += 2"
+  done
+
+  echo "Waiting for build to start..."
+  timeout=0
+  until [[ $(curl -s http://localhost/job/containers-quickstarts/job/${AGENT}/lastBuild/api/json --user admin:${token} | jq -r '.building') == "true" ]]
+  do
+    if [[ ${timeout} -gt 60 ]]
+    then
+      echo "Timed out waiting for build to start..."
+      exit 1
+    fi
+    sleep 2
+    let "timeout += 2"
+  done 
+
+  echo "Build in progress..."
+  timeout=0
+  while [[ $(curl -s http://localhost/job/containers-quickstarts/job/${AGENT}/lastBuild/api/json --user admin:${token} | jq -r '.building') == "true" ]]
+  do
+    if [[ ${timeout} -gt 300 ]]
+    then
+      echo "Timed out waiting for build to finish..."
```

Comment:
would be nice if these timeout breakouts dumped some logs. happy for that to be solved in another pr.

Reply:
Agreed, I'll add the build logs to the build timeout but not sure what we could/should log for the job start timeout?